### PR TITLE
Datagen packet size

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

Adding packet size option to most generators (multiblade excluded for now)

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: prototype2/gdgem/generators/nmxgen_hits.cpp [hunk 1]
```diff
@@ -97,6 +115,6 @@ int main(int argc, char *argv[]) {
     usleep(Settings.SpeedThrottle * 1000);
   }
 
-  return 0;
+  return (EXIT_SUCCESS);
```
Comment (amues) [inline prototype2/gdgem/generators/nmxgen_hits.cpp]: The use of `return` here seems inconsistent with the use of `exit` in line 99.

File: prototype2/gdgem/generators/nmxgen_hits.cpp [hunk 2]
```diff
@@ -69,11 +77,21 @@ int main(int argc, char *argv[]) {
 
   size_t SentPackets{0}; // counter to determine when to break
   for (;;) {
-    int readsz = file.read(buffer);
-    if (readsz > 0 && (Settings.NumberOfPackets == 0 || SentPackets < Settings.NumberOfPackets)) {
-      DataSource.send(buffer, readsz);
-      Bytes += readsz;
-      Packets++;
+    if (Settings.NumberOfPackets == 0 || SentPackets < Settings.NumberOfPackets) {
+      int readsz = file.read(buffer);
```
Comment (amues) [inline prototype2/gdgem/generators/nmxgen_hits.cpp]: The declaration of `buffer` sets its size to 9000, which is compatible but has to be synchronised manually with the value of 9000 in _DumpFile.h:49_, should it ever change. Using a constant in this case could ensure the numbers stay the same.

File: prototype2/gdgem/generators/nmxgen_hits.cpp [hunk 3]
```diff
@@ -1,48 +1,48 @@
...
-
 #include <libs/include/TSCTimer.h>
 #include <libs/include/Timer.h>
 
 #include <gdgem/generators/ReaderHits.h>
 #include <libs/include/Socket.h>
-#include <unistd.h>
 // GCOVR_EXCL_START
+
 static constexpr int TscMHz {2900};
```
Comment (amues) [inline prototype2/gdgem/generators/nmxgen_hits.cpp]: Where does this number come from?

File: prototype2/gdgem/generators/nmxgen_readouts.cpp [hunk 4]
```diff
@@ -1,41 +1,41 @@
 /** Copyright (C) 2016, 2017 European Spallation Source ERIC */
```
Comment (amues) [inline prototype2/gdgem/generators/nmxgen_readouts.cpp]: The comments from _nmxgen_hits.cpp_ apply to this file as well.

File: prototype2/multigrid/generators/mggen_readouts.cpp [hunk 5]
```diff
@@ -1,20 +1,48 @@
-/** Copyright (C) 2016, 2017 European Spallation Source ERIC */
```
Comment (amues) [inline prototype2/multigrid/generators/mggen_readouts.cpp]: The comments from _nmxgen_hits.cpp_ apply to this file as well.

File: prototype2/gdgem/generators/nmxgen_hits.cpp [hunk 6]
```diff
@@ -67,17 +75,22 @@ int main(int argc, char *argv[]) {
...
-  size_t SentPackets{0}; // counter to determine when to break
   for (;;) {
-    int readsz = file.read(buffer);
-    if (readsz > 0 && (Settings.NumberOfPackets == 0 || SentPackets < Settings.NumberOfPackets)) {
-      DataSource.send(buffer, readsz);
-      Bytes += readsz;
-      Packets++;
+    int BytesToSend = file.read(buffer);
+    int BytesSent = 0;
+    if (BytesToSend > 0 && (Settings.NumberOfPackets == 0 || Packets < Settings.NumberOfPackets)) {
```
Comment (amues) [inline prototype2/gdgem/generators/nmxgen_hits.cpp]: `Packets` is being reset in line 106, which makes it unsuitable for this usage.

**Comment by mortenjc** [comment]
ok, thanks for reviewing this thoroughly and finding several problems. These should now be fixed
  * added constexpr for buffer size
  * validating buffersize with h5 file chunksize
  * removing unreachable return
  * using EXIT_FAILURE and EXIT_SUCCESS
  * replace printf with fmt::format
  * fixing logic bug for exit condition.

**Pull request merged**